### PR TITLE
chore(jest): [RO-26531] Migrated test runner to Jest

### PR DESCRIPTION
**Pull request by otrreeves**
### Ticket
https://opentable.atlassian.net/browse/RO-26531

### Summary
This PR will update the test runner from Mocha to Jest

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      functions: 84,
+      lines: 89,
+      statements: 89
+    }
+  },
+  moduleFileExtensions: ['js','json'],
+  rootDir: '.',
+  testEnvironment: 'node',
+  testMatch: ['**/*.spec.js'],
+  testResultsProcessor: 'jest-teamcity-reporter',
```
Comment (acolchado) [inline jest.config.js:18]: This is not needed since it runs in Github actions.

**Review by acolchado — COMMENTED** [review]
Looks good overall, but tests are failing in the GH env and the teamcity reporter is not needed.

File: jest.config.js (hunk 1 above)
Comment (otrreeves) [reply jest.config.js]: Removed

**Review by acolchado — APPROVED** [review]
Lgtm

**Pull request merged**